mlflow/gateway/app.py

```diff
@@ -2,63 +2,44 @@
 from fastapi.responses import RedirectResponse
 import logging
 import os
-from pathlib import Path
-from typing import List, Union
+from typing import Any, Optional, Dict
 
 from mlflow.version import VERSION
+from mlflow.exceptions import MlflowException
 from mlflow.gateway.constants import MLFLOW_GATEWAY_HEALTH_ENDPOINT
 from mlflow.gateway.config import (
     Route,
     RouteConfig,
     RouteType,
+    GatewayConfig,
     _load_route_config,
 )
 from mlflow.gateway.schemas import chat, completions, embeddings
-from .providers import get_provider
-from .schemas import chat, completions, embeddings
+from mlflow.gateway.providers import get_provider
 
 _logger = logging.getLogger(__name__)
 
 MLFLOW_GATEWAY_CONFIG = "MLFLOW_GATEWAY_CONFIG"
 
-# Configured and initialized Gateway Routes state index
-ACTIVE_ROUTES: List[Route] = []
```

Comment:
If we have a variable like this, we always have to be careful with mutation, so I removed it.